receiver/memcachedreceiver/scraper.go

```diff
@@ -191,32 +159,29 @@ func (r *memcachedScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 		parsedHit, okHit := r.parseInt("incr_hits", stats.Stats["incr_hits"])
 		parsedMiss, okMiss := r.parseInt("incr_misses", stats.Stats["incr_misses"])
 		if okHit && okMiss {
-			r.addToDoubleMetric(hitRatio, attributes, calculateHitRatio(parsedHit, parsedMiss), now)
+			r.mb.RecordMemcachedOperationHitRatioDataPoint(now, calculateHitRatio(parsedHit, parsedMiss), "increment")
 		}
 
 		attributes = pcommon.NewMap()
 		attributes.Insert(metadata.A.Operation, pcommon.NewValueString("decrement"))
 		parsedHit, okHit = r.parseInt("decr_hits", stats.Stats["decr_hits"])
 		parsedMiss, okMiss = r.parseInt("decr_misses", stats.Stats["decr_misses"])
 		if okHit && okMiss {
-			r.addToDoubleMetric(hitRatio, attributes, calculateHitRatio(parsedHit, parsedMiss), now)
+			r.mb.RecordMemcachedOperationHitRatioDataPoint(now, calculateHitRatio(parsedHit, parsedMiss), "decrement")
 		}
 
 		attributes = pcommon.NewMap()
 		attributes.Insert(metadata.A.Operation, pcommon.NewValueString("get"))
 		parsedHit, okHit = r.parseInt("get_hits", stats.Stats["get_hits"])
 		parsedMiss, okMiss = r.parseInt("get_misses", stats.Stats["get_misses"])
 		if okHit && okMiss {
-			r.addToDoubleMetric(hitRatio, attributes, calculateHitRatio(parsedHit, parsedMiss), now)
+			r.mb.RecordMemcachedOperationHitRatioDataPoint(now, calculateHitRatio(parsedHit, parsedMiss), "get")
 		}
 	}
-	return md, nil
-}
 
-func initMetric(ms pmetric.MetricSlice, mi metadata.MetricIntf) pmetric.Metric {
-	m := ms.AppendEmpty()
-	mi.Init(m)
-	return m
+	// What do I emit? The generated code expects ResourceOptions but the metadata specified no resources
```

Comment:
@dmitryax since metadata.yaml specifies no Resources, what should I pass to the Emit function?

Reply:
Nothing. Emit has optional arguments only. 

I'll probably update the code generator to make Emit without any arguments if there are no resource attributes in the yaml file to avoid confusion